docs/design/coreclr/botr/logging.md

```diff
@@ -0,0 +1,70 @@
+
+For basic scenarios, you can use the [`dotnet-counters`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-counters) tool to monitor events and performance counters the runtime reports through EventPipe. For example, to collect the default counters from a project while running it, you could use `dotnet-counters collect -- dotnet run`.
+
+You can also use the [`dotnet-trace`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-trace) tool to capture EventPipe events from the runtime on the fly, either saving them to a file or sending them to your console. It provides command line options you can use to filter events by severity level or based on keywords. For example, to capture high-severity GC events only from a project while running it, you could use `dotnet-trace collect --clreventlevel error --clrevents gc -- dotnet run`.
+
+Please see the linked documentation pages for more information on how to use each of the above tools. They have an extensive set of options and some helpful features like time limits, attaching to existing processes, and configurable output formatting.
+
+For advanced scenarios, on Windows you can also use standard ETW tools like [WPR](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-recorder) to capture events from EventPipe, and on Linux you can use [LTTNG](https://lttng.org/).
+
+To emit your very own EventPipe events from within the C++ part of the runtime, you can use the `FireEtwXXX` APIs, or their convenience wrappers within the `ETW::` C++ namespace. You likely will want to create new type(s) of events instead of using existing ones. The events are generated by the [`genEventing.py`](https://github.com/dotnet/runtime/blob/main/src/coreclr/scripts/genEventing.py) script based on the definitions in [`ClrEtwAll.man`](https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/ClrEtwAll.man). **(FIXME: Is the process for adding one documented somewhere?)**
+
+To emit EventPipe events from C# you can use the [`System.Diagnostics.Tracing.EventSource`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventsource-getting-started) class, by defining your own event types derived from it. See its documentation for examples and more details.
+
+StressLog
+------------
+The StressLog is a circular buffer inside the runtime process that usually does not escape, and most StressLog messages are enabled in retail builds, which makes it useful for troubleshooting issues with GC or other subsystems in production scenarios. To enable it, you can set the `DOTNET_StressLog` environment variable to `1`, and you can configure it by setting the `DOTNET_LogFacility`, `DOTNET_LogFacility2` and `DOTNET_LogLevel` environment variables. If you are in a situation where you can't attach a debugger to the process, the `DOTNET_StressLogFilename` environment variable can be used to write the StressLog's contents to a file instead.
```

Comment:
There are more env variables to control stresslog. The two definitely worth mentioning are `DOTNET_StressLogSize` and `DOTNET_TotalStressLogSize` can be used to configure the size of per-thread circular buffer and cap the total size of all circular buffers in the process.

For example, here is my default combo to diagnose GC hole type crashes: https://github.com/dotnet/runtime/issues/45557#issuecomment-738547930

Reply:
it'd be great to list all the env vars you can set first (and give it some value like what's in Jan's comment) and then explain what kind of values they can take.

Reply:
I added more detailed information on some of the specific stress log and traditional log environment variables in each of their sections, along with a section that lists all the log levels and facilities. Do you think this is sufficient? I can expand more on any section you'd like to see include more detail or if I've missed an environment variable you think is important.

Reply:
what I meant was the following, first list these env vars 
```
set DOTNET_StressLog=1
set DOTNET_LogLevel=6
set DOTNET_LogFacility=0x00080001
set DOTNET_StressLogSize=0x2000000
set DOTNET_TotalStressLogSize=0x40000000
...

optional -
set DOTNET_LogFlushFile=1
set DOTNET_LogFileAppend=1
...
```
some of these are very self explanatory like `DOTNET_StressLog`, and this is something you can just copy and paste (instead of having to each one from the text that follows and copy) and either don't need to change at all or only change very little. then you can explain what the non self explanatory ones like  -

```
DOTNET_StressLogSize
configures how large the per-thread buffer for log messages is

DOTNET_TotalStressLogSize 
configures a process-wide cap to avoid exhausting memory when you have many threads

DOTNET_LogLevel
see [put a link to the "Log Facilities and Levels" section here]

...
```
personally I find this format a lot easier to read than putting all of them in one or a few block of text but of course that's a bit subjective.

Reply:
How's the updated version of the stresslog and traditional logging sections? Thanks for the suggestion to rework it this way, I think it's more readable.